### ISSUE TITLE
OSGi: neither stats.json nor token file are not read correctly in OSGi

### ISSUE THREAD

**Issue by denis-anisimov** [vaadin/flow#9146]
When I try to register a Vaadin servlet using HTTP whiteboard I  cannot get it working out of the box in NPM mode because
of two issues:
- token file (by default) is read via classloader resource (`DeploymentConfigurationFactory::getResourceFromClassloader`).
- stats.json file (by default) is also read via classloader resource `FrontendUtils::getStatsAssetsByChunkName` .

In both cases the classloader which tries to find a resource is the classloader which loaded the `flow-server`  bundle.
And this classloader may not access the resource from another bundle.
And it's in fact just incorrect try to access the resource that way in OSGi: `flow-server` may be deployed as a standalone bundle and there can be a numerous of WAB deployed . So there will be a number of `stats.json` files with the same path in different bundles. So it's not possible to choose (and in fact all of them should be chosen per WAB).

The way to load those resources should be changed somehow via OSGi services : the resources should be loaded somehow per WAB.
In plain Web app `flow-server`  is a part of the Web app and it's just loaded by the same classloader as the whole app and that's why there is no such issue.
In OSGi `flow-server` doesn't belong to the Web application anyhow (it's a "singleton").

To be able to make WAB working in NPM mode I have to implement a workaround: 
```
private static class FixedVaadinServlet extends VaadinServlet {
        @Override
        public void init(ServletConfig servletConfig) throws ServletException {
            super.init(servletConfig);

            getService().setClassLoader(getClass().getClassLoader());
        }

        @Override
        protected DeploymentConfiguration createDeploymentConfiguration(
                Properties initParameters) {
            initParameters.remove(
                    DeploymentConfigurationFactory.DEV_MODE_ENABLE_STRATEGY);
            initParameters.put(InitParameters.SERVLET_PARAMETER_PRODUCTION_MODE,
                    true);
            DeploymentConfiguration config = new DefaultDeploymentConfiguration(
                    getClass(), initParameters) {
                @Override
                public boolean isProductionMode() {
                    return true;
                }

                @Override
                public boolean isStatsExternal() {
                    return true;
                }

                @Override
                public String getExternalStatsUrl() {
                    return "/VAADIN/config/stats.json";
                }
            };

            return config;
        }

    }
```

And also I have to register somehow "/VAADIN/config/stats.json" URL . E.g. via Http Service:
```
@Activate
    void activate(BundleContext ctx) throws NamespaceException {
        httpService.registerResources("/VAADIN/config/stats.json",
                "/META-INF/VAADIN/config/stats.json",
                httpService.createDefaultHttpContext());
    }

    @Reference
    void setHttpService(HttpService service) {
        this.httpService = service;
    }

    void unsetHttpService(HttpService service) {
        this.httpService = null;
    }

```

**Comment by pleku** [comment]
Acceptance Criteria
- [ ] the project base doesn't have any hack for the stats.json nor the token file

**Comment by denis-anisimov** [comment]
Done in the feature branch.

**Comment by mg-dev-fis** [comment]
Should be part of official release, available in LTS.

**Comment by pleku** [comment]
Hi @mg-dev-fis that is planned for 2021, but for now it is only available in Vaadin 19.0.0.alpha1.
My bet would be at this point that it would be available in 14.6 which will not land before the summer 2021 but that is still well before the EOL for Vaadin 8 (Feb/2022).
